### ISSUE TITLE
`no-unused-modules` erroneously flags `as default` exports

### ISSUE THREAD

**Issue by builtbyjavi** [benmosher/eslint-plugin-import#3216]
## Issue Description

The `no-unused-modules` rule incorrectly reports exports as unused when they use the `export { X as default }` syntax, even when the default export is properly imported and used by other modules.

This appears to be a bug in how the plugin resolves default exports that are created via the re-export syntax rather than inline default export declarations.

## Minimal Reproduction

**test-export.ts**
```typescript
export function testHandler() {
  return 'result';
}

export { testHandler as default };
```

**test-import.ts**
```typescript
import handler from './test-export';
console.log(handler());
```

**Current behavior**: `eslint-plugin-import` reports `export function testHandler` in `test-export.ts` as unused.

**Expected behavior**: The rule should recognize that:
1. `testHandler` is re-exported as the default export
2. The default export is imported and used in `test-import.ts`
3. Therefore, neither the named export nor the re-export as default should be flagged as unused

## Configuration

```json
{
  "import/no-unused-modules": ["error", { "unusedExports": true }]
}
```

## Environment

- **eslint-plugin-import version**: 2.31.0
- **Node version**: 22.13.1
- **npm version**: 10.9.2

**Comment by oconnorjoseph** [comment]
+1

**Comment by LaishGlenberg** [comment]
Very frustrating that this still hasn't been fixed, there are dozens of issues and fixes dating back years for this exact same problem, why does it keep coming back up? Can we please get this PR merged soon? I spent my entire afternoon trying to get this plugin to work.
